src/payment/strategies/stripe-upe/stripe-upe-payment-strategy.ts

```diff
@@ -185,6 +194,47 @@ export default class StripeUPEPaymentStrategy implements PaymentStrategy {
         throw error;
     }
 
+    private _mapStripeAddress(address?: Address): AddressOptions {
+        if (address) {
+            const {
+                city,
+                countryCode: country,
+                postalCode,
+                stateOrProvince: state,
+            } = address;
+
+            return { city, country, postal_code: postalCode, state };
+        }
+
+        throw new MissingDataError(MissingDataErrorType.MissingBillingAddress);
+    }
+
+    private _mapStripePaymentData(): StripeConfirmPaymentData {
+        const billingAddress = this._store.getState().billingAddress.getBillingAddress();
+        const address = {
+            address:  this._mapStripeAddress(billingAddress),
+        };
+
+        const email = billingAddress?.email;
+
+        if (!this._stripeElements) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+
+        return {
+            elements: this._stripeElements,
+            redirect: StripeStringConstants.IF_REQUIRED,
+            confirmParams: {
+                payment_method_data: {
+                    billing_details: {
+                        email,
```

Comment:
Here `email` if I understand correctly can be undefined? And address can be similar to `{ }`. Do we want to type check before assigning? Because if email is undefined and address is empty then this could be 
`billing_details: { email: undefined }`

Reply:
I added type check

Reply:
Although I don’t think these are needed since we always have information for these values in the payment step, and additionally, stripe throws an error if one of these fields is undefined.